packages/core/src/browser/xhrProxy.ts

```diff
@@ -69,85 +80,81 @@ export function resetXhrProxy() {
+      return
+    }
+    this._datadog_xhr = {
+      ...this._datadog_xhr,
+      startTime: relativeNow(),
+      startClocks: clocksNow(),
+      isAborted: false,
+    }
 
-  XMLHttpRequest.prototype.send = function (this: BrowserXHR) {
-    callMonitored(() => {
-      if (this._datadog_xhr) {
-        const xhrPendingContext = this._datadog_xhr as XhrStartContext &
-          Pick<XhrCompleteContext, 'startTime' | 'isAborted'>
-        xhrPendingContext.startTime = relativeNow()
-        xhrPendingContext.startClocks = clocksNow()
-        xhrPendingContext.isAborted = false
-
-        const originalOnreadystatechange = this.onreadystatechange
-
-        this.onreadystatechange = function () {
-          if (this.readyState === XMLHttpRequest.DONE) {
-            // Try to report the XHR as soon as possible, because the XHR may be mutated by the
-            // application during a future event. For example, Angular is calling .abort() on
-            // completed requests during a onreadystatechange event, so the status becomes '0'
-            // before the request is collected.
-            callMonitored(reportXhr)
-          }
-
-          if (originalOnreadystatechange) {
-            originalOnreadystatechange.apply(this, arguments as any)
-          }
-        }
-
-        let hasBeenReported = false
-
-        const reportXhr = () => {
-          if (hasBeenReported) {
-            return
-          }
-          hasBeenReported = true
-
-          const xhrCompleteContext: XhrCompleteContext = {
-            ...xhrPendingContext,
-            duration: elapsed(xhrPendingContext.startClocks.timeStamp, timeStampNow()),
-            response: this.response as string | undefined,
-            status: this.status,
-          }
-
-          onRequestCompleteCallbacks.forEach((callback) => callback(xhrCompleteContext))
-        }
-
-        this.addEventListener('loadend', monitor(reportXhr))
-
-        beforeSendCallbacks.forEach((callback) => callback(xhrPendingContext, this))
+    let hasBeenReported = false
+    const originalOnreadystatechange = this.onreadystatechange
+    const onreadystatechange = function (this: BrowserXHR) {
+      if (this.readyState === XMLHttpRequest.DONE) {
+        // Try to report the XHR as soon as possible, because the XHR may be mutated by the
+        // application during a future event. For example, Angular is calling .abort() on
+        // completed requests during a onreadystatechange event, so the status becomes '0'
+        // before the request is collected.
+        onEnd()
       }
-    })
 
-    return originalXhrSend.apply(this, arguments as any)
-  }
+      if (originalOnreadystatechange) {
+        originalOnreadystatechange.apply(this, arguments as any)
+      }
+    }
 
-  XMLHttpRequest.prototype.abort = function (this: BrowserXHR) {
-    callMonitored(() => {
-      if (this._datadog_xhr) {
-        this._datadog_xhr.isAborted = true
+    const onEnd = monitor(() => {
```

Comment:
Does it change the behavior for the case where the xhr is aborted because the same instance reopen a request?

Reply:
If you refer to the scenario 1 of the [RFC](https://docs.google.com/document/d/1nLE_EoAdqOYsYbtRej0qUruO63S6D-WUaWmQbj_RYnw/edit#heading=h.36igkj90d6fd), no the behaviour is still the same as the current version.